endoreg_db/services/video_import.py

```
@@ -158,181 +272,557 @@ def _create_or_retrieve_video_instance(self):
+    def _get_processor_roi_info(self) -> Tuple[Optional[List[List[Dict[str, Any]]]], Optional[Dict[str, Any]]]:
+        """Get processor ROI information for masking."""
+        endoscope_data_roi_nested = None
+        endoscope_image_roi = None
+
+        video = self._require_current_video()
+
+        try:
+            video_meta = getattr(video, "video_meta", None)
+            processor = getattr(video_meta, "processor", None) if video_meta else None
+            if processor:
+                assert isinstance(processor, EndoscopyProcessor), "Processor is not of type EndoscopyProcessor"
```

Pylint (4.0.1)

[error] Using variable 'EndoscopyProcessor' before assignment

(E0601)